crates/circuit/src/circuit_data.rs

```diff
@@ -186,6 +281,44 @@ impl CircuitData {
         self.qubits.len()
     }
 
+    /// The list of registered :class:`.ClassicalRegisters` instances.
+    ///
+    /// .. warning::
+    ///
+    ///     Do not modify this list yourself.  It will invalidate the :class:`CircuitData` data
+    ///     structures.
```

Comment:
`CircuitData` isn't public, we wrap it in :class:`.QuantumCircuitData` which is not in the docs apparently but referred to as the return type of `QuantumCircuit.data`

```suggestion
    ///     Do not modify this list yourself.  It will invalidate/corrupt :attr:`.data` for this circuit.
```